### PR TITLE
Added http status codes

### PR DESCRIPTION
**Pull request by diogo464**
I didn't find status codes in the library so I scrapped them from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

### CODE REVIEW

File: include/cpr/status_codes.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#ifndef _CPR_STATUS_CODES
+#define _CPR_STATUS_CODES
+#include <cstdint>
+namespace cpr {
+namespace status {
+constexpr std::uint32_t HTTP_100_CONTINUE = 100;
```
Comment (tstack) [inline include/cpr/status_codes.h]: The current Response class uses int32_t for the status code, so this should probably match up.

File: include/cpr/status_codes.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+constexpr std::uint32_t HTTP_101_SWITCHING_PROTOCOL = 101;
```
Comment (tstack) [inline include/cpr/status_codes.h]: I'm finding it a little unusual to include the code number in the constant name, are you okay with removing it (e.g. make this HTTP_SWITCHING_PROTOCOL)?

Comment (diogo464) [reply include/cpr/status_codes.h]: I had them like that because I was used to Django but I've removed them now.

**Comment by tstack** [comment]
thanks!

**Comment by mtangoo** [comment]
Great addition. I have found that status reason important especially to the user. There is this code[1] here under CC0. Should we add it to library so that when we get status code we can request reason to display to user for example?

[1] https://github.com/j-ulrich/http-status-codes-cpp

**Pull request merged**